containers/src/Kokkos_Vector.hpp

```diff
@@ -147,6 +147,72 @@ class vector : public DualView<Scalar*, LayoutLeft, Arg1Type> {
 
   void clear() { _size = 0; }
 
+  void insert(const_iterator it, const value_type& val) { insert(it, 1, val); }
+
+  void insert(const_iterator it, size_type count, const value_type& val) {
+    DV::sync_host();
+    DV::modify_host();
+    if (it < begin() || it >= end())
+      Kokkos::abort("Kokkos::vector::insert : invalid insert iterator");
+
+    ptrdiff_t start = it - begin();
+    auto org_size   = size();
+    resize(size() + count);
+    auto org_view =
+        subview(DV::h_view, pair<ptrdiff_t, ptrdiff_t>(start, org_size));
+    auto new_view = subview(DV::h_view, pair<ptrdiff_t, ptrdiff_t>(
+                                            start + count, org_size + count));
+
+    for (ptrdiff_t i = org_view.extent(0) - 1; i >= 0; i--) {
```

Comment:
Why go backwards?

Reply:
because I otherwise overwrite values I still need to copy?
```c++
for(int i=5; i<a.size()-1;i++)
  a[i+1]=a[i];
```
write `a[6]` before reading `a[6]` again to assign it to `a[7]` and so on.
I.e. I just filled the rest of `a` with the original value from `a[5]`

Reply:
oh, right. Of course.  A comment to that effect would probably be helpful, though